auth/email-link.html

```diff
@@ -0,0 +1,240 @@
+<!DOCTYPE html>
+<!--
+Copyright (c) 2016 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<html>
+<head>
+  <meta charset=utf-8 />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Email Link Authentication Example</title>
+
+  <!-- Material Design Theming -->
+  <link rel="stylesheet" href="https://code.getmdl.io/1.1.3/material.orange-indigo.min.css">
+  <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
+  <script defer src="https://code.getmdl.io/1.1.3/material.min.js"></script>
+
+  <link rel="stylesheet" href="main.css">
+
+  <!-- Import and configure the Firebase SDK -->
+  <!-- These scripts are made available when the app is served or deployed on Firebase Hosting -->
+  <!-- If you do not serve/host your project using Firebase Hosting see https://firebase.google.com/docs/web/setup -->
+  <script src="/__/firebase/4.11.0/firebase-app.js"></script>
+  <script src="/__/firebase/4.11.0/firebase-auth.js"></script>
+  <script src="/__/firebase/init.js"></script>
+
+  <script type="text/javascript">
+
+    /**
+     * Handles the sign in button press.
+     */
+    function toggleSignIn() {
+      // Disable the sign-in button during async sign-in tasks.
+      document.getElementById('quickstart-sign-in').disabled = true;
+
+      if (firebase.auth().currentUser) {
+        // [START signout]
+        firebase.auth().signOut().catch(function(error) {
+          // Handle Errors here.
+          var errorCode = error.code;
+          var errorMessage = error.message;
+          // [START_EXCLUDE]
+          handleError(error);
+          // [END_EXCLUDE]
+        });
+        // [END signout]
+      } else {
+        var email = document.getElementById('email').value;
+        // Sending email with sign-in link.
+        // [START authwithemail]
+        var actionCodeSettings = {
+          // URL you want to redirect back to. The domain (www.example.com) for this URL
+          // must be whitelisted in the Firebase Console.
+          'url': window.location.href, // Here we redirect back to this same page.
+          'handleCodeInApp': true // This must be true.
+         };
+
+        firebase.auth().sendSignInLinkToEmail(email, actionCodeSettings).then(function() {
+          // Save the email locally so you don’t need to ask the user for it again if they open
+          // the link on the same device.
+          window.localStorage.setItem('emailForSignIn', email);
```

Comment:
One advantage of using a cookie is that it will expire naturally after some point. localStorage will save this indefinitely if the action is not completed. The downside is that cookies are sent to server. Should we at least mention this?